src/types/StateCollection.ts

```diff
@@ -0,0 +1,6 @@
+import { Entity } from "./Entity"
+
+export interface StateCollection<A extends Entity> {
+  entities: { [key: string]: A }
+  result: readonly Entity["id"][]
```

Comment:
⛏ Should this be `A["id"]`?

Reply:
I can't get the types to work like you would think....  not sure why... let's look at this together.